load.go

```diff
@@ -187,3 +187,43 @@ func ToKindProps[T KindProperties](v cue.Value) (T, error) {
 
 	return *props, nil
 }
+
+// ToDef takes a cue.Value expected to represent a kind of the category
+// specified by the type parameter and populates a Def from the CUE value.
+// The cue.Value in Def.V will be the unified value of the parameter cue.Value
+// and the kindsys CUE kind (Core, Custom, Composable).
+func ToDef[T KindProperties](v cue.Value) (Def[T], error) {
```

Comment:
This function is needed by codegen to ensure that the `joinSchema` changes are actually unified with the lineage in the `cue.Value`, otherwise just using `ToKindProps` you load all the unified properties, but still have the old lineage when you make a call to `Bind<X>`

Reply:
This seems almost exactly the same as [`ToKindProps`](https://github.com/grafana/kindsys/blob/5df3e256ebc05d223e068ce1e12143cb72967974/load.go#L154-L156) - why can't this just wrap that? Or just have the caller use that directly?

fwiw this is one area where i've thought for a while that there may be a simplification opportunity. I have an inkling that we can get rid of the `Def` type in favor of something more unified with the different `Kind` interfaces.

Reply:
It's basically `ToKindProps` that exposes the unified `cue.Value`, which `ToKindProps` doesn't do--since `Custom` alters the lineage, we need the unified `cue.Value` when binding the lineage, so we can't just use `ToKindProps`.